src/test/java/org/springframework/data/couchbase/domain/AirportJsonValuedObject.java

```diff
@@ -0,0 +1,39 @@
+package org.springframework.data.couchbase.domain;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import org.springframework.data.annotation.PersistenceCreator;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+
+public class AirportJsonValuedObject {
+
+	public AirportJsonValuedObject() {}
+
+	@PersistenceCreator
+	@JsonCreator
```

Comment:
How is this new feature related to `@JsonCreater`? Shouldn't Spring Data code use `@PersistenceCreator` instead?

Reply:
Aside from my hack with storing the enum constants in a map, there should aways be an @JsonCreator method if one wants to deserialize an object from what the output serialized by the @JsonValue.  What I have in the JsonValueConverter.read(), if the method can't figure out how to deserialize with either an @JsonCreator method or the enum constants hack, it throws an exception which is caught by the calling readValue() which goes on to deserialize by the regular spring-data process.  It first looks for an @PersistenceCreator constructor to use. (If it doesn't find one, it creates the object with a no-args constructor and then sets the args).  
I used this for testing both, first one, then the other manually.  When both are present, the JsonValueConverter gets the first try - with the @JsonCreator.  I should have made two objects, one with @PersistenceCreateor, one with @JsonCreator.

Reply:
Got it! Thanks for the explanation.